demux/demux_lavf.c

```diff
@@ -466,9 +466,12 @@ static int lavf_check_file(demuxer_t *demuxer, enum demux_check check)
         }
     }
 
+    // HLS streams seems to be not well tagged, so matching mime type is not
+    // enough. Strip URL parameters and match extension.
+    bstr ext = bstr_get_ext(bstr_split(bstr0(priv->filename), "?#", NULL));
     AVProbeData avpd = {
         // Disable file-extension matching with normal checks, except for HLS
-        .filename = av_match_ext(priv->filename, "m3u8,m3u") ||
+        .filename = !bstrcasecmp0(ext, "m3u8") || !bstrcasecmp0(ext, "m3u") ||
                     check <= DEMUX_CHECK_REQUEST ? priv->filename : "",
```

Comment:
maybe we should pass "dummy.m3u" if *we're* convinced that it's HLS so that ffmpeg doesn't get any funny ideas

Reply:
I would like to have parity with what `ffmpeg.c` is doing. If there is a bug upstream it should be fixed there instead of workaround it here.

They have fancier url parsing, but the proposed here should be enough to catch the streams and let lavf process it further.